src/lib/markbind/src/lib/markdown-it/patches/markdown-it-attrs-nunjucks.js

```diff
@@ -0,0 +1,78 @@
+/*
+ Patch for markdown-it-attrs and nunjuck's usage of {% ... %}
+ */
+
+const mdAttrsUtils = require('markdown-it-attrs/utils');
+
+mdAttrsUtils.hasDelimiters = function (where, options) {
+
+  if (!where) {
+    throw new Error('Parameter `where` not passed. Should be "start", "middle", "end" or "only".');
+  }
+
+  /**
+   * @param {string} str
+   * @return {boolean}
+   */
+  return function (str) {
+    // we need minimum three chars, for example {b}
+    let minCurlyLength = options.leftDelimiter.length + 1 + options.rightDelimiter.length;
+    if (!str || typeof str !== 'string' || str.length < minCurlyLength) {
+      return false;
+    }
+
+    function validCurlyLength (curly) {
+      let isClass = curly.charAt(options.leftDelimiter.length) === '.';
+      let isId = curly.charAt(options.leftDelimiter.length) === '#';
+      return (isClass || isId)
+        ? curly.length >= (minCurlyLength + 1)
+        : curly.length >= minCurlyLength;
+    }
+
+    let start, end, slice, nextChar;
+    let rightDelimiterMinimumShift = minCurlyLength - options.rightDelimiter.length;
+    switch (where) {
+    case 'start':
+      // first char should be {, } found in char 2 or more
+      slice = str.slice(0, options.leftDelimiter.length);
+      start = slice === options.leftDelimiter ? 0 : -1;
+      end = start === -1 ? -1 : str.indexOf(options.rightDelimiter, rightDelimiterMinimumShift);
+      // check if next character is not one of the delimiters
+      nextChar = str.charAt(end + options.rightDelimiter.length);
+      if (nextChar && options.rightDelimiter.indexOf(nextChar) !== -1) {
+        end = -1;
+      }
+      break;
+
+    case 'end':
+      // last char should be }
+      start = str.lastIndexOf(options.leftDelimiter);
+      end = start === -1 ? -1 : str.indexOf(options.rightDelimiter, start + rightDelimiterMinimumShift);
+      end = end === str.length - options.rightDelimiter.length ? end : -1;
+      break;
+
+    case 'only':
+      // '{.a}'
+      slice = str.slice(0, options.leftDelimiter.length);
+      start = slice === options.leftDelimiter ? 0 : -1;
+      slice = str.slice(str.length - options.rightDelimiter.length);
+      end = slice === options.rightDelimiter ? str.length - options.rightDelimiter.length : -1;
+      break;
+    }
+
+    /*
+     Simple patch here - abort if the delimiters wrap around % %
+     */
+    const isCharAfterStartPercent = str.charAt(start + 1) === '%';
```

Comment:
*the only functional changes in this file are these 5 lines

Reply:
Just to confirm, this won't break the other nunjucks templating syntax that use `{% ...%}` right?

Reply:
yup, if anything it should 'fix' it!

e.g.
```
{% for x in y %}
{{ x }}
{% endfor %}
```

this previously works (and still does) because nunjucks rendering is the first step and long precedes markdown rendering, so the `{% for/endfor %}` is 'erased' long before markdown-it-attrs touches it.
It dosen't work in the case of {% raw/endraw %} because it is still present at the time of markdown rendering though

So this should give more flexibility in future restructuring of the rendering order (if ever needed)

Reply:
That makes sense, thanks for clarifying!